src/hypothesis/strategies.py

```diff
@@ -803,21 +803,51 @@ def dm_func(denom):
 
 @cacheable
 @defines_strategy
-def decimals(min_value=None, max_value=None):
+def decimals(
+    min_value=None, max_value=None, allow_nan=None, allow_infinity=None
+):
     """Generates instances of decimals.Decimal.
 
-    If min_value is not None then all generated values are no less than
-    min_value.
+    - If min_value is not None, all values will be >= min_value.
+    - If max_value is not None, all values will be <= max_value.
+    - If min_value or max_value is not None, it is an error to enable
+      allow_nan.
+    - If both min_value and max_value are not None, it is an error to enable
+      allow_infinity.
 
-    If max_value is not None then all generated values are no greater than
-    max_value.
+    Where not explicitly ruled out by the bounds, all of infinity, -infinity
+    and NaN are possible values generated by this strategy.
 
     """
     check_valid_bound(min_value, 'min_value')
     check_valid_bound(max_value, 'max_value')
     check_valid_interval(min_value, max_value, 'min_value', 'max_value')
 
-    floats_strategy = floats(min_value=min_value, max_value=max_value)
+    # The `allow_nan` and `allow_infinity` flags were added to the `decimals()`
+    # strategy after `min_value` and `max_value`.  To preserve backward-
+    # compatibility, if these flags are defaulted then we select an appropriate
+    # value based on the other flags supplied to this strategy.
```

Comment:
Good point, I think I was originally going based on your comment in #183:

> In order to preserve the current behaviour these should default to True.

In the original commit, I did that in the function signature itself (`def decimals(..., allow_nan=True, allow_infinity=True)`), which breaks any existing code. I'll tweak this comment.